packages/api/cms-api/src/access-log/access-log.interceptor.ts

```diff
@@ -0,0 +1,59 @@
+import { CallHandler, ExecutionContext, Injectable, Logger, NestInterceptor } from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+
+/** TODOs
+ * Docs are missing
+ * Api password check
+ */
+const DAM_IMAGE_PATH = "/dam/images/:hash/:fileId";
+const DAM_FILE_PATH = "/dam/files/:hash/:fileId";
+
+@Injectable()
+export class AccessLoggerInterceptor implements NestInterceptor {
+    protected readonly logger = new Logger(AccessLoggerInterceptor.name);
+
+    intercept(context: ExecutionContext, next: CallHandler) {
+        const requestType = context.getType().toString();
+
+        let message = "";
+        const additionalRequestData = [];
+        let ignored = false;
+
+        if (requestType === "graphql") {
+            const graphqlExecutionContext = GqlExecutionContext.create(context);
+            const graphqlContext = graphqlExecutionContext.getContext();
+
+            additionalRequestData.push(`ip: ${graphqlContext.req.ip}`);
+            const user = graphqlContext.req.user;
+            if (user) additionalRequestData.push(`user: ${user.id} (${user.name})`);
+
+            const gqlArgs = graphqlExecutionContext.getArgs();
+            const gqlInfo = graphqlExecutionContext.getInfo();
+
+            if (gqlInfo.operation.operation === "mutation") {
+                delete gqlArgs["input"];
+                delete gqlArgs["data"];
+            }
+
+            message = `operationType: ${gqlInfo.parentType} - operationName: ${gqlInfo.operation.name.value} - resolver function: ${
+                gqlInfo.fieldName
+            } - args: ${JSON.stringify(gqlArgs)}`;
```

Comment:
Do we need the resolver name? Shouldn't this be clear from the operation name?

Reply:
Yes it should, but then you always have to look explicitly into the admin code and find the resolver (if you can already see that the "error" happens in the api). if the resolver name is in the log you could jump directly into the api function.

But thats my opinion, I can also remove the resolver entry.

Reply:
The operation name could be set so anything by an attacker, I would keep both